c2corg_ui/static/js/search.js

```diff
@@ -163,44 +172,58 @@ app.SearchController = function($scope, $compile, $attrs, apiUrl, gettextCatalog
 
 /**
  * @param {string} type The document type.
- * @param {string} title The category title.
  * @return {TypeaheadDataset} A data set.
  * @private
  */
-app.SearchController.prototype.createDataset_ = function(type, title) {
+app.SearchController.prototype.createDataset_ = function(type) {
   var bloodhoundEngine = this.createAndInitBloodhound_(type);
   return /** @type {TypeaheadDataset} */({
+    contents : type,
     source: bloodhoundEngine.ttAdapter(),
     display: function(doc) {
       return doc.label;
     },
     limit: 20,
     templates: {
       header: (function() {
-        return '<div class="header ' + title + '">' +
-            this.gettextCatalog_.getString(title) + '</div>';
+        return '<div class="header" dataset="' + type + '">' + this.gettextCatalog_.getString(type) + '</div>';
       }).bind(this),
+      footer: function() {
+        return '<p class="suggestion-more"><a href="/' + type + '" class="green-text">+ see more results</a></p>';
+      },
       suggestion: function(doc) {
-        return '<p>' + doc.label + '</p>';
-      }
-      // Note: The templates for `notFound` and `pending` are not set, because
-      // they would be shown for each dataset.
+        var document = {
+          'query' : doc['_query'],
+          'elevation' : doc['elevation'] || doc['elevation_max'],
+          'waypoint_type' : doc['waypoint_type'],
+          'activities' : doc['activities'],
+          'date_end' : doc['date_end'],
+          'doctype' : doc['documentType']
+        };
+        return this.compile_('<app-suggestion label="' + doc['label'] + '" doc=\'' + JSON.stringify(document) + '\' class="tt-suggestion"></app-suggestion>')(this.scope_);
```

Comment:
I'm nearly there, but I still encounter a problem. I can access the passed document through scope.$parent['document'], I extend the directive's scope by the document from the parent, but the template doesn't seem to compile correctly and not a single value is evaluated.

My directive 

```
return {
    restrict: 'E',
    scope: true, 

    link: function(scope, element, attrs) {

      var document = scope.$parent['doc']
      angular.extend(scope, scope.$parent)
     // if i console.log(scope), I have all of the needed attributes of the document

      element.html(template(document));
      $compile(element.contents())(scope);  // I could use scope.$parent as well, but it doesn't change
    }
  }
```

and template like `... '<span ng-if="elevation_max">{{::elevation_max}}m</span>' ....`